ecs-logging-core/src/main/java/co/elastic/logging/EcsJsonSerializer.java

```diff
@@ -36,7 +36,7 @@ public class EcsJsonSerializer {
     private static final TimestampSerializer TIMESTAMP_SERIALIZER = new TimestampSerializer();
     private static final ThreadLocal<StringBuilder> messageStringBuilder = new ThreadLocal<StringBuilder>();
     private static final String NEW_LINE = System.getProperty("line.separator");
-    private static final Pattern NEW_LINE_PATTERN = Pattern.compile("\\n");
+    private static final Pattern NEW_LINE_PATTERN = Pattern.compile("\\R");
```

Comment:
`\R` has been introduced in Java 8 so this won't work with older Java versions which this library is compatible with.

You could either use the equivalent of `\R` (`\u000D\u000A|[\u000A\u000B\u000C\u000D\u0085\u2028\u2029]`) or the more simple alternative `"\\r?\\n"` which should cover most relevant cases (ignores Mac OS 9 style `\r`, though).

Reply:
ah okay, was not aware that < Java 8 is supported still. Yes then make it right. The pattern I like as it is simple to read is \r\n|\n|\r - covers all cases for a single newline. Will update